src/components/asset-list/RecyclerAssetList2/profile-header/ProfileActionButtonsRow.tsx

```diff
@@ -0,0 +1,419 @@
+import Clipboard from '@react-native-community/clipboard';
+import lang from 'i18n-js';
+// @ts-expect-error - JS module
+import { IS_TESTING } from 'react-native-dotenv';
+import * as React from 'react';
+import { PressableProps } from 'react-native';
+import Animated, {
+  useAnimatedStyle,
+  useDerivedValue,
+  withSpring,
+} from 'react-native-reanimated';
+import { ButtonPressAnimation } from '@/components/animations';
+import { enableActionsOnReadOnlyWallet } from '@/config';
+import {
+  AccentColorProvider,
+  Box,
+  Column,
+  Columns,
+  Inset,
+  Stack,
+  Text,
+  useColorMode,
+} from '@/design-system';
+import { maybeSignUri } from '@/handlers/imgix';
+import {
+  CurrencySelectionTypes,
+  ExchangeModalTypes,
+  NetworkTypes,
+} from '@/helpers';
+import {
+  useAccountProfile,
+  useAccountSettings,
+  useLatestCallback,
+  usePersistentDominantColorFromImage,
+  useSwapCurrencyHandlers,
+  useWallets,
+} from '@/hooks';
+import { delayNext } from '@/hooks/useMagicAutofocus';
+import { useNavigation } from '@/navigation';
+import { useTheme } from '@/theme';
+import { showActionSheetWithOptions, watchingAlert } from '@/utils';
+import Routes from '@rainbow-me/routes';
+import { FloatingEmojis } from '@/components/floating-emojis';
+import showWalletErrorAlert from '@/helpers/support';
+import { analytics } from '@/analytics';
+import ContextMenu from '@/components/native-context-menu/contextMenu';
+import { useRecoilState } from 'recoil';
+import { addressCopiedToastAtom } from '@/screens/WalletScreen';
+
+export const ProfileActionButtonsRowHeight = 80;
+
+export function ProfileActionButtonsRow() {
+  // ////////////////////////////////////////////////////
+  // Account
+  const { accountColor, accountImage } = useAccountProfile();
+
+  // ////////////////////////////////////////////////////
+  // Colors
+
+  const { result: dominantColor, state } = usePersistentDominantColorFromImage(
+    maybeSignUri(accountImage ?? '') ?? ''
+  );
+
+  const { colors } = useTheme();
+  let accentColor = colors.white;
+  if (accountImage) {
+    accentColor = dominantColor || colors.white;
+  } else if (typeof accountColor === 'number') {
+    accentColor = colors.avatarBackgrounds[accountColor];
+  }
+
+  // ////////////////////////////////////////////////////
+  // Animations
+
+  const hasAvatarLoaded = !accountImage && accountColor !== undefined;
+  const hasImageColorLoaded = state === 2 || state === 3;
+  const hasLoaded = hasAvatarLoaded || hasImageColorLoaded;
+
+  const scale = useDerivedValue(() => {
```

Comment:
I would just use scale directly in the `useAnimatedStyle` and added `scale` in the dependency array.

Or use `useReactiveSharedValue`

Reply:
Good point!